archive_converter.py

```diff
@@ -13,8 +13,57 @@
 # limitations under the License.
 
 import email
+import os
 from setup_gmail import Message
 
+class ArchiveMessageIndex(object):
+    def __init__(self):
+        self._lookup_index = {} # Maps MessageKey to Thread
```

Comment:
add typing info here

```py
self._lookup_index : Dict[str, Message] = {}
```

But actually, I think a name change could be useful.
We generally "lookup" things in an "index", so the name seems a bit redundant atm, given it's not returning an "index" but a Message (according to type annotation below)

```py
self._key_to_thread_start : Dict[str, Message] = {}
```

perhaps?